lte/gateway/c/oai/tasks/s1ap/s1ap_mme_encoder.c

```diff
@@ -19,437 +19,142 @@
-
     case S1ap_ProcedureCode_id_UEContextRelease:
-      return s1ap_mme_encode_ue_context_release_command(
-          message_p, buffer, length);
-
-    case S1ap_ProcedureCode_id_Paging:
-      return s1ap_mme_encode_paging(message_p, buffer, length);
-
-    case S1ap_ProcedureCode_id_UEContextModification:
-      return s1ap_mme_encode_ue_context_modification_request(
-          message_p, buffer, length);
-
     case S1ap_ProcedureCode_id_E_RABSetup:
-      return s1ap_mme_encode_e_rab_setup(message_p, buffer, length);
-
+    case S1ap_ProcedureCode_id_E_RABModify:
     case S1ap_ProcedureCode_id_E_RABRelease:
-      return s1ap_mme_encode_e_rab_release_command(message_p, buffer, length);
-
+    case S1ap_ProcedureCode_id_HandoverResourceAllocation:
+    case S1ap_ProcedureCode_id_MMEStatusTransfer:
+    case S1ap_ProcedureCode_id_Paging:
     case S1ap_ProcedureCode_id_MMEConfigurationTransfer:
-      return s1ap_mme_encode_mme_configuration_transfer(
-          message_p, buffer, length);
+      break;
 
     default:
-      OAILOG_DEBUG(
+      OAILOG_NOTICE(
           LOG_S1AP, "Unknown procedure ID (%d) for initiating message_p\n",
-          (int) message_p->procedureCode);
-      break;
+          (int) pdu->choice.initiatingMessage.procedureCode);
+      *buffer = NULL;
+      *length = 0;
+      return -1;
   }
 
-  return -1;
+  memset(&res, 0, sizeof(res));
+  res = asn_encode_to_new_buffer(
+      NULL, ATS_ALIGNED_CANONICAL_PER, &asn_DEF_S1ap_S1AP_PDU, pdu);
+  *buffer = res.buffer;
+  *length = res.result.encoded;
+  return 0;
 }
 
 //------------------------------------------------------------------------------
 static inline int s1ap_mme_encode_successfull_outcome(
-    s1ap_message* message_p, uint8_t** buffer, uint32_t* length) {
-  switch (message_p->procedureCode) {
+    S1ap_S1AP_PDU_t* pdu, uint8_t** buffer, uint32_t* length) {
+  asn_encode_to_new_buffer_result_t res = {NULL, {0, NULL, NULL}};
+  DevAssert(pdu != NULL);
+
+  switch (pdu->choice.successfulOutcome.procedureCode) {
     case S1ap_ProcedureCode_id_S1Setup:
-      return s1ap_mme_encode_s1setupresponse(message_p, buffer, length);
-    case S1ap_ProcedureCode_id_Reset:
-      return s1ap_mme_encode_resetack(message_p, buffer, length);
     case S1ap_ProcedureCode_id_PathSwitchRequest:
-      return s1ap_mme_encode_pathswitchreqack(message_p, buffer, length);
+    case S1ap_ProcedureCode_id_HandoverPreparation:
+    case S1ap_ProcedureCode_id_HandoverCancel:
+    case S1ap_ProcedureCode_id_Reset:
+    case S1ap_ProcedureCode_id_E_RABModificationIndication:
+      break;
 
     default:
       OAILOG_DEBUG(
           LOG_S1AP,
           "Unknown procedure ID (%d) for successfull outcome message\n",
-          (int) message_p->procedureCode);
-      break;
+          (int) pdu->choice.successfulOutcome.procedureCode);
+      *buffer = NULL;
+      *length = 0;
+      return -1;
   }
 
-  return -1;
+  memset(&res, 0, sizeof(res));
```

Comment:
res is already initialized with zero and NULL; Do we need to use memset here?

Reply:
No, removed memset.